From d5f081966351f80c818cd97974b0b61eebd10fef Mon Sep 17 00:00:00 2001
From: Chin-Ting Kuo <chin-ting_kuo@aspeedtech.com>
Date: Thu, 25 Sep 2025 14:36:44 +0800
Subject: [PATCH 32/32] riscv: aspeed: Enable SSP/TSP reset bits in reset mask

When EXTRST# reset is triggered, SSP and TSP coprocessors
will be reset by default. This behavior is also applied to
WDT reset.

Signed-off-by: Chin-Ting Kuo <chin-ting_kuo@aspeedtech.com>
Change-Id: I7240b674b3b1a7c5b12991cbc48b6e0628d38b7c
(cherry picked from commit 412268d9d02e1cee6a38bdbff05bb6f470b5ed54)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/riscv/include/asm/arch-ast2700/extrst.h | 2 +-
 board/aspeed/ibex_ast2700/wdt.c              | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/riscv/include/asm/arch-ast2700/extrst.h b/arch/riscv/include/asm/arch-ast2700/extrst.h
index 2f89926fe7..32353a8a99 100644
--- a/arch/riscv/include/asm/arch-ast2700/extrst.h
+++ b/arch/riscv/include/asm/arch-ast2700/extrst.h
@@ -11,7 +11,7 @@
 #include <asm/arch/platform.h>
 #include <linux/types.h>
 
-#define SCU0_EXTRST_MASK_1_VAL	0x8207e771
+#define SCU0_EXTRST_MASK_1_VAL	0x8207ff71
 #define SCU0_EXTRST_MASK_2_VAL	0x000003f6
 
 #define SCU1_EXTRST_MASK_1_VAL	0x000093ec
diff --git a/board/aspeed/ibex_ast2700/wdt.c b/board/aspeed/ibex_ast2700/wdt.c
index e267836bad..f20bcbc675 100644
--- a/board/aspeed/ibex_ast2700/wdt.c
+++ b/board/aspeed/ibex_ast2700/wdt.c
@@ -27,14 +27,14 @@ int wdt_init(void)
 		for (idx = 0; idx < 8; idx++) {
 			wdt_base_addr = ASPEED_WDT_BASE + idx * 0x80;
 			/* SoC reset mask */
-			wdt_writel(0x8207e771, (void *)(wdt_base_addr + WDT_RST_MASK_1));
+			wdt_writel(0x8207ff71, (void *)(wdt_base_addr + WDT_RST_MASK_1));
 			wdt_writel(0x000003f6, (void *)(wdt_base_addr + WDT_RST_MASK_2));
 			wdt_writel(0x000093ec, (void *)(wdt_base_addr + WDT_RST_MASK_3));
 			wdt_writel(0x40303803, (void *)(wdt_base_addr + WDT_RST_MASK_4));
 			wdt_writel(0x003e0000, (void *)(wdt_base_addr + WDT_RST_MASK_5));
 
 			/* SW reset mask */
-			wdt_writel(0x8207e771, (void *)(wdt_base_addr + WDT_SW_RST_MASK_1));
+			wdt_writel(0x8207ff71, (void *)(wdt_base_addr + WDT_SW_RST_MASK_1));
 			wdt_writel(0x000003f6, (void *)(wdt_base_addr + WDT_SW_RST_MASK_2));
 			wdt_writel(0x000093ec, (void *)(wdt_base_addr + WDT_SW_RST_MASK_3));
 			wdt_writel(0x40303803, (void *)(wdt_base_addr + WDT_SW_RST_MASK_4));
-- 
2.34.1

